docs/docs/prompt-management/03-prompt-management-sdk.mdx

```diff
@@ -0,0 +1,500 @@
+---
+title: "Manage Prompt with the SDK"
+---
+
+import Image from "@theme/IdealImage";
+
+In this guide, we'll explore all actions performed through the SDK: creating variants, committing changes, deploying changes, and fetching configurations. For a simpler introduction, please refer to the [prompt mangement tutorial](/tutorials/sdk/manage-prompts-with-SDK).
+
+Before starting, we recommend to get acquainted with how versioning works in Agenta. You can find more details on the [concepts page](/concepts/concepts), or read the quick summary below.
+
+:::info Versioning in Agenta
+
+{" "}
+
+<Image
+  style={{ display: "block", margin: "16px 0px 16px 0px" }}
+  img={require("/images/prompt_management/taxonomy-concepts.png")}
+  alt="Taxonomy of concepts in Agenta"
+  loading="lazy"
+/>
+Agenta follows a structure similar to **git** for prompt versioning. Instead of a
+single commit history, you can create multiple branches—called **variants**. Each
+variant represents a distinct approach or solution you're exploring.
+
+To move from experimentation to deployment, Agenta uses **environments** like
+**development**, **staging**, and **production**. You can deploy specific versions of your **variants**
+to these **environments**, controlling what gets tested and what goes live.
+
+The workflow of deploying something to production is therefore as follows:
+
+1. (optionally) Create a new **variant** (branch)
+2. Commit a change to that **variant**
+3. Deploy that **commit** (**variant**/**version**) to an **environment**
+
+:::
+
+## Setup
+
+Before using the SDK, you need to initialize it using the `ag.init()` method.
+
+```python
+import agenta as ag
+
+# Initialize the SDK with your API key
+os.environment["AGENTA_API_KEY"] = "xxx" # Only needs setting in oss
+os.environment["AGENTA_HOST"] = "https://cloud.agenta.ai" # default value no need setting explicitly
+ag.init()
+```
+
+## Creating a new prompt
+
+Each prompt in agenta is a unique application. Currently creating a prompt is only available in the web UI. To create a new prompt, just click on create a new prompt and select whether it's a chat or completion prompt in the web UI.
+
+## Committing changes
+
+### Creating a New Variant
+
+To create a new variant, use the `VariantManager.create_variant` method.
+
+```python
+
+from agenta import Prompt
```

Comment:
Seeing that the attributes of Prompt would be subject to change... This needs to be something that the user would create themselves, and not imported from Agenta.